bitcoin/src/util/psbt/mod.rs

```diff
@@ -1689,6 +1712,100 @@ mod tests {
+                        ).unwrap(),
+                        vout: 0,
+                    },
+                    sequence: Sequence::ENABLE_LOCKTIME_NO_RBF,
+                    ..Default::default()
+                }],
+                output: vec![
+                    TxOut {
+                        value: output_0_val,
+                        ..Default::default()
+                    },
+                    TxOut {
+                        value: output_1_val,
+                        ..Default::default()
+                    },
+                ],
+            },
+            xpub: Default::default(),
+            version: 0,
+            proprietary: BTreeMap::new(),
+            unknown: BTreeMap::new(),
+
+            inputs: vec![Input {
+                non_witness_utxo: Some(Transaction {
+                    version: 1,
+                    lock_time: absolute::PackedLockTime::ZERO,
+                    input: vec![TxIn {
+                        previous_output: OutPoint {
+                            txid: Txid::from_hex(
+                                "e567952fb6cc33857f392efa3a46c995a28f69cca4bb1b37e0204dab1ec7a389",
+                            ).unwrap(),
+                            vout: 1,
+                        },
+                        sequence: Sequence::MAX,
+                       ..Default::default()
+                    },
+                    TxIn {
+                        previous_output: OutPoint {
+                            txid: Txid::from_hex(
+                                "b490486aec3ae671012dddb2bb08466bef37720a533a894814ff1da743aaf886",
+                            ).unwrap(),
+                            vout: 1,
+                        },
+                        sequence: Sequence::MAX,
+                        ..Default::default()
+                    }],
+                    output: vec![
+                        TxOut {
+                            value: prev_output_val,
+                            ..Default::default()
+                        },
+                        TxOut {
+                            value: 190303501938,
+                            ..Default::default()
+                        },
+                    ],
+                }),
+                ..Default::default()
+            },],
+            outputs: vec![
+                Output {
+                    ..Default::default()
+                },
+                Output {
+                    ..Default::default()
+                },
+            ],
+        };
+        assert_eq!(t.fee().unwrap(), prev_output_val - (output_0_val + output_1_val));
+        // no previous output
+        let mut t2 = t.clone();
+        t2.inputs[0].non_witness_utxo = None;
+        assert!(t2.fee().is_err());
+        // invalid fee
+        let mut t3 = t.clone();
+        t3.unsigned_tx.output[0].value = prev_output_val;
+        assert!(t3.fee().is_err());
+
+    }
+
```

Comment:
Please add a test for overflow case when you fix it. This is a very important one.

Reply:
done